components/eventing-controller/pkg/handlers/nats.go

```diff
@@ -313,20 +309,14 @@ func (n *Nats) getCallback(subKeyPrefix, subscriptionName string) nats.MsgHandle
 			Period:   n.defaultSubsConfig.DispatcherRetryPeriod,
 		}
 		if result := n.doWithRetry(traceCtxWithCE, retryParams, ce); !cev2.IsACK(result) {
-			n.emitDeliveryMetric(subscriptionName, ce.Type(), sinkValue, http.StatusBadRequest)
+			n.metricsCollector.RecordDeliveryPerSubscription(subscriptionName, ce.Type(), sinkValue, http.StatusBadRequest)
```

Comment:
why don't you pass the `sink` variable here instead of `sinkValue`?

Reply:
done 👍